File: dcp-solutions/2024-11-01/solution.rb
```ruby
class Queue
  def initialize
    @stack1 = []
    @stack2 = []
  end

  # Enqueue an element onto the queue
  def enqueue(element)
    @stack1.push(element)
  end

  # Dequeue an element from the queue
  def dequeue
    return if @stack2.empty?

    element = @stack2.pop
    while !@stack1.empty?
      @stack2.push(@stack1.pop)
    end

    element
  end
end
```

File: dcp-solutions/2024-11-01/solution.py
```python
class Queue:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def enqueue(self, x: int) -> None:
        self.stack1.append(x)

    def dequeue(self) -> int:
        if not self.stack2:
            while self.stack1:
                self.stack2.append(self.stack1.pop())
        return self.stack2.pop()

```